lib/vrl/compiler/src/expression/assignment.rs

```diff
@@ -422,7 +422,7 @@ where
 // -----------------------------------------------------------------------------
 
 #[derive(Debug, Clone)]
-pub(crate) struct Details {
+pub struct Details {
     pub type_def: TypeDef,
     pub value: Option<Value>,
 }
```

Comment:
I'm having a hard time understanding why we need to make this assignment-related type public.

Reply:
Good catch. It's not needed now that we don't insert enrichment tables as a variable any more.

Reply:
Ah yeah, of course. _That's_ why this was needed before 👍